californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -137,6 +144,12 @@
 	/** The matcher which matches incoming responses, akcs and rsts an exchange */
 	private final Matcher matcher;
 
+	/** Serializer to convert messages to datagrams. */
+	private final Serializer serializer;
```

Comment:
No state, but using an instance allow only CoapEndpoint constructor to care about TPC vs UDP, as it instantiates the right serializer and parser. Otherwise every reference to serializer and parser needs to be wrapped by an if.